src/Categories/Category/Construction/Monoids.agda

```diff
@@ -0,0 +1,74 @@
+{-# OPTIONS --without-K --safe #-}
+open import Categories.Category.Core
+open import Categories.Category.Monoidal.Core
+
+module Categories.Category.Construction.Monoids {o ℓ e} {𝒞 : Category o ℓ e} (C : Monoidal 𝒞) where
+
+open import Level
+
+open import Categories.Functor using (Functor)
+open import Categories.Object.Monoid C
+
+open Category 𝒞
+open Monoidal C
+open HomReasoning
+open Monoid using (η; μ)
+open Monoid⇒
+
+Monoids : Category (o ⊔ ℓ ⊔ e) (ℓ ⊔ e) e
+Monoids = record
+  { Obj = Monoid
+  ; _⇒_ = Monoid⇒
+  ; _≈_ = λ f g → arr f ≈ arr g
+  ; id = λ {A} → record
+    { arr = id
+    ; preserves-μ = begin
+      id ∘ μ A
+        ≈⟨ identityˡ ⟩
+      μ A
+        ≈⟨ sym identityʳ ⟩
+      μ A ∘ id
+        ≈⟨ (refl ⟩∘⟨ sym (Functor.identity ⊗)) ⟩
+      μ A ∘ id ⊗₁ id
+        ∎
+    ; preserves-η = identityˡ
+    }
+  ; _∘_ = λ {A B C} f g → record
+    { arr = arr f ∘ arr g
+    ; preserves-μ = begin
+      (arr f ∘ arr g) ∘ μ A
+        ≈⟨ assoc ⟩
```

Comment:
look at the push/pull combinators, as they embody this exact idea of assoc then operate on one side.

Reply:
Thank you for the tip! I've shortened pretty much all the proofs now.